jupyter_client/manager.py

```diff
@@ -476,7 +489,10 @@ async def _async_shutdown_kernel(self, now: bool = False, restart: bool = False)
         # Stop monitoring for restarting while we shutdown.
         self.stop_restarter()
 
-        await ensure_async(self.interrupt_kernel())
+        # If the kernel has already started, interrupt it to give it a
+        # chance to clean up.
+        if self.has_kernel:
+            await ensure_async(self.interrupt_kernel())
```

Comment:
Nice - make _this_ call to interrupt() _conditional_ on start status.  Good catch.

Reply:
👍🏼